### PR TITLE
Removed the need to have the local REDOX secret in configuration

### PR DESCRIPTION
**Pull request by carlosfelix2**
This PR removes the need to have the local REDOX secret in configuration.  This was generating an error on the Git Leaks pre-hook.

## Changes
- Set a default secret key in the code only for local environment.

### Testing
- [x] Tested locally?
- [ ] Ran `quickTest all`?
- [x] Ran `./prime test` against local Docker ReportStream container?
- [ ] Downloaded a file from `http://localhost:7071/api/download`?
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

### CODE REVIEW

File: prime-router/build.gradle.kts [hunk 1]
```diff
@@ -343,7 +343,6 @@ tasks.azureFunctionsRun {
             "POSTGRES_PASSWORD" to dbPassword,
             "POSTGRES_URL" to dbUrl,
             "PRIME_ENVIRONMENT" to "local",
-            "REDOX_SECRET" to "some_secret",
```
Comment (td-usds) [inline prime-router/build.gradle.kts:346]: Note that just yanking this does not obviate the need to suppress the warning; the nightly run checks the history, not just the state of master as is. In other words, unless you supress this one, the nightly gitleaks run will continue to flag this.

File: prime-router/docker-compose.yml [hunk 2]
```diff
@@ -27,8 +27,6 @@ services:
       # postgresql comes from the build environment; it is accessible because we are connected to the build network (see "networks")
       - POSTGRES_URL=jdbc:postgresql://postgresql:5432/prime_data_hub
       - PRIME_ENVIRONMENT=local
-      - REDOX_SECRET=some_secret
```
Comment (td-usds) [inline prime-router/docker-compose.yml:30]: see above

**Comment by td-usds** [comment]
I don't know if this is the branch on which you keep seeing the terraform formatting errors, but when I jumped on to this branch, and then git merged master, I no longer see those errors on git commit

**Review by jimduff-usds — APPROVED** [review]
lgtm

**Review by td-usds — CHANGES_REQUESTED** [review]
Note that you will still have to provide suppressions for the lines you remove since GitLeaks scans history, not 'current state'.
Please include those as well since they are preventing others from merging things in (and so I'd like it in sooner rather than later).

**Comment by carlosfelix2** [comment]
> Note that you will still have to provide suppressions for the lines you remove since GitLeaks scans history, not 'current state'.
> Please include those as well since they are preventing others from merging things in (and so I'd like it in sooner rather than later).

You will need to explain how to add suppression.

**Comment by ronaldheft-usds** [comment]
> > Note that you will still have to provide suppressions for the lines you remove since GitLeaks scans history, not 'current state'.
> > Please include those as well since they are preventing others from merging things in (and so I'd like it in sooner rather than later).
> 
> You will need to explain how to add suppression.

It's documented here: https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/docs/allowlist-gitleaks-false-positives.md

**Comment by carlosfelix2** [comment]
> > > Note that you will still have to provide suppressions for the lines you remove since GitLeaks scans history, not 'current state'.
> > > Please include those as well since they are preventing others from merging things in (and so I'd like it in sooner rather than later).
> > 
> > 
> > You will need to explain how to add suppression.
> 
> It's documented here: https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/docs/allowlist-gitleaks-false-positives.md

thanks!

**Review by td-usds — APPROVED** [review]

**Review by td-usds — APPROVED** [review]

**Pull request merged**